torch/csrc/Module.cpp

```diff
@@ -429,6 +429,54 @@ PyObject* THPModule_fromDLPack(PyObject* _unused, PyObject* data) {
   END_HANDLE_TH_ERRORS
 }
 
+PyObject* THPModule_isHooksAvailable_wrap(PyObject* _unused, PyObject* arg) {
+  THPUtils_assert(
+      THPUtils_checkString(arg),
+      "_is_hooks_available expects a str, "
+      "but got %s",
+      THPUtils_typename(arg));
+  std::string device_type = THPUtils_unpackString(arg);
+  if (device_type == "cuda" && at::hasCUDA()) {
+    Py_RETURN_TRUE;
+  }
+  if (device_type == "xpu" && at::hasXPU()) {
+    Py_RETURN_TRUE;
+  }
+  if (device_type == "mps" && at::hasMPS()) {
+    Py_RETURN_TRUE;
+  }
+  // add more available device types here
+  Py_RETURN_FALSE;
+}
+
+PyObject* THPModule_getDeviceCount_wrap(PyObject* _unused, PyObject* arg) {
+  THPUtils_assert(
+      THPUtils_checkString(arg),
+      "_get_device_count expects a str, "
+      "but got %s",
+      THPUtils_typename(arg));
+  std::string device_type = THPUtils_unpackString(arg);
+  if (device_type == "xpu" && at::hasXPU()) {
+    return THPUtils_packUInt64(at::detail::getXPUHooks().device_count());
+  }
+  // add more available device types here
+  return THPUtils_packUInt64(0);
```

Comment:
This function is written to be backend agnostic, but it seems wrong to silent say "device count = 0" for any other backend here. Can you update it to:
(1) error for other backends
(2) add a comment saying that other backends should extend the logic here?